### PR TITLE
Remove obsolete BraveVPNLinkSubscriptionAndroidUIStudy

### PR DESCRIPTION
**Pull request by bsclifton**

### CODE REVIEW

**Comment by bsclifton** [comment]
NOTE: should be merged AFTER https://github.com/brave/brave-variations/pull/535

**Review by kjozwiak — APPROVED** [review]
Uplift into `production` approved after deliberating with @brave/uplift-approvers. QA has verified the PR on `main` (staging) as per https://github.com/brave/brave-variations/pull/457#issuecomment-1462811197, https://github.com/brave/brave-variations/pull/457#issuecomment-1462861073, https://github.com/brave/brave-variations/pull/457#issuecomment-1462892777https://github.com/brave/brave-variations/pull/457#issuecomment-1463921918.

**Pull request merged**
